internal/db/schema/migrations/oss/postgres/70/02_sessions.up.sql

```diff
@@ -0,0 +1,189 @@
+begin;
+
+  -- Replaces trigger from 56/02_add_data_key_foreign_key_references.up.sql
+  drop trigger cancel_session_with_null_fk on session;
+  create or replace function cancel_session_with_null_fk() returns trigger
+  as $$
+  begin
+    -- Note that we need each of these to run in case
+    -- more than one of them is null.
+    if new.auth_token_id is null then
+      perform cancel_session(new.public_id);
+    end if;
+    if new.project_id is null then
+      -- Setting the key_id to null will allow the scope
+      -- to cascade delete its keys.
+      new.key_id = null;
+      perform cancel_session(new.public_id);
+    end if;
+    if new.target_id is null then
+      perform cancel_session(new.public_id);
+    end if;
+    if new.user_id is null then
+      perform cancel_session(new.public_id);
+    end if;
+    return new;
+  end;
+  $$ language plpgsql;
+  
+  create trigger cancel_session_with_null_fk before update of auth_token_id, project_id, target_id, user_id on session
+    for each row execute procedure cancel_session_with_null_fk();
+
+  create table session_target_address (
+    session_id wt_public_id primary key,
+    target_id wt_public_id,
+    -- the following foreign key constraint is set to null on a delete
+    -- because we want to be able to invoke a trigger to call cancel_session
+    -- when the references column is deleted
+    -- rather than canceling a session for any reason that may cause a row to be deleted.
+    constraint target_address_fkey foreign key (target_id)
+        references target_address (target_id)
+        on delete set null
+        on update cascade,
+    constraint session_fkey foreign key (session_id)
+        references session (public_id)
+        on delete cascade
+        on update cascade,
+    constraint session_id_target_id_uq
```

Comment:
```suggestion
    constraint session_target_address_session_id_target_id_uq
```
Unique constraint names are global so they must be prefixed with the table name.